File: mobile/ios/Sources/PolicyWarningBanner.swift
import SwiftUI

/// The acknowledge gate, shown in place of the chat composer.
///
/// A moderator issued a warning about sharing contact details and the server is
/// refusing this user's messages (HTTP 409) until they confirm they have read
/// it. Nothing else notifies them — no email, no push — so this banner IS the
/// delivery, which is why it replaces the composer rather than sitting above it:
/// a notice you can ignore while still typing is not a gate.
///
/// Used by both `ChatView` (booking threads) and `ConversationChatView`
/// (pre-booking threads).
struct PolicyWarningBanner: View {
    let text: String
    let isAcknowledging: Bool
    let onAcknowledge: () -> Void

    var body: some View {
        VStack(alignment: .leading, spacing: 10) {
            Label(L.t("chat.warningTitle"), systemImage: "exclamationmark.triangle.fill")
                .font(.subheadline.weight(.bold))
                .foregroundStyle(Color.qkBurgundy)

            Text(text)
                .font(.callout)
                .foregroundStyle(Color.qkInk)
                .fixedSize(horizontal: false, vertical: true)
                .frame(maxWidth: .infinity, alignment: .leading)

            Button(action: onAcknowledge) {
                ZStack {
                    if isAcknowledging {
                        ProgressView().tint(.white)
                    } else {
                        Text(L.t("chat.warningAck")).font(.subheadline.weight(.bold))
                    }
                }
                .frame(maxWidth: .infinity)
                .padding(.vertical, 12)
                .background(Color.qkBurgundy)
                .foregroundStyle(.white)
                .clipShape(RoundedRectangle(cornerRadius: 14, style: .continuous))
            }
            .disabled(isAcknowledging)
        }
        .padding(14)
        .background(Color.white)
        .clipShape(RoundedRectangle(cornerRadius: 18, style: .continuous))
        .overlay(
            RoundedRectangle(cornerRadius: 18, style: .continuous)
                .stroke(Color.qkBurgundy.opacity(0.35), lineWidth: 1)
        )
        .padding(.horizontal, 12)
        .padding(.top, 8)
        .padding(.bottom, 10)
    }
}

File: mobile/android/app/src/main/java/com/quickin/app/ui/PolicyWarningBanner.kt
package com.quickin.app.ui

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.quickin.app.ui.theme.Burgundy

// Matches the file-private constant used by the chat screens.
private val ErrorRed = Color(0xFFB3261E)

/**
 * The acknowledge gate, shown in place of the chat composer.
 *
 * A moderator issued a warning about sharing contact details and the API is
 * refusing this user's messages (HTTP 409) until they confirm they have read it.
 * Nothing else notifies them — no email, no push — so this banner IS the
 * delivery, which is why it replaces the input bar rather than sitting above it:
 * a notice you can ignore while still typing is not a gate.
 *
 * Used by both [ChatScreen] (booking threads) and the pre-booking thread screen.
 */
@Composable
fun PolicyWarningBanner(
    text: String,
    isAcknowledging: Boolean,
    onAcknowledge: () -> Unit,
) {
    Surface(
        color = Color.White,
        shape = RoundedCornerShape(16.dp),
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = 12.dp, vertical = 8.dp),
    ) {
        Column(Modifier.padding(14.dp)) {
            Text(
                "A note from QuickIn",
                color = ErrorRed,
                fontSize = 14.sp,
                fontWeight = FontWeight.Bold,
                modifier = Modifier.padding(bottom = 6.dp),
            )
            Text(
                text,
                fontSize = 14.sp,
                modifier = Modifier.padding(bottom = 12.dp),
            )
            Button(
                onClick = onAcknowledge,
                enabled = !isAcknowledging,
                shape = RoundedCornerShape(14.dp),
                colors = ButtonDefaults.buttonColors(containerColor = Burgundy, contentColor = Color.White),
                modifier = Modifier.fillMaxWidth(),
            ) {
                if (isAcknowledging) {
                    CircularProgressIndicator(color = Color.White, strokeWidth = 2.dp, modifier = Modifier.padding(2.dp))
                } else {
                    Text("I understand", fontWeight = FontWeight.Bold)
                }
            }
        }
    }
}
